// styled-material/cjs/typography.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.Caption = exports.Body2 = exports.Body = exports.Subhead2 = exports.Subhead = exports.Title = exports.Headline = exports.Display1 = exports.Display2 = exports.Display3 = exports.Display4 = undefined;

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _templateObject = _taggedTemplateLiteral(['\n  font-size: 112px;\n  font-weight: 300;\n  line-height: 1;\n  letter-spacing: -0.04em;\n'], ['\n  font-size: 112px;\n  font-weight: 300;\n  line-height: 1;\n  letter-spacing: -0.04em;\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n  font-size: 56px;\n  font-weight: 400;\n  line-height: 1.35;\n  letter-spacing: -0.02em;\n'], ['\n  font-size: 56px;\n  font-weight: 400;\n  line-height: 1.35;\n  letter-spacing: -0.02em;\n']),
    _templateObject3 = _taggedTemplateLiteral(['\n  font-size: 45px;\n  font-weight: 400;\n  line-height: 48px;\n'], ['\n  font-size: 45px;\n  font-weight: 400;\n  line-height: 48px;\n']),
    _templateObject4 = _taggedTemplateLiteral(['\n  font-size: 34px;\n  font-weight: 400;\n  line-height: 40px;\n'], ['\n  font-size: 34px;\n  font-weight: 400;\n  line-height: 40px;\n']),
    _templateObject5 = _taggedTemplateLiteral(['\n  font-size: 24px;\n  font-weight: 400;\n  line-height: 32px;\n  -moz-osx-font-smoothing: grayscale;\n'], ['\n  font-size: 24px;\n  font-weight: 400;\n  line-height: 32px;\n  -moz-osx-font-smoothing: grayscale;\n']),
    _templateObject6 = _taggedTemplateLiteral(['\n  font-size: 20px;\n  font-weight: 500;\n  line-height: 1;\n'], ['\n  font-size: 20px;\n  font-weight: 500;\n  line-height: 1;\n']),
    _templateObject7 = _taggedTemplateLiteral(['\n  margin: 0;\n  font-size: 16px;\n  font-weight: 400;\n  line-height: 24px;\n'], ['\n  margin: 0;\n  font-size: 16px;\n  font-weight: 400;\n  line-height: 24px;\n']),
    _templateObject8 = _taggedTemplateLiteral(['\n  margin: 0;\n  font-size: 16px;\n  font-weight: 400;\n  line-height: 28px;\n'], ['\n  margin: 0;\n  font-size: 16px;\n  font-weight: 400;\n  line-height: 28px;\n']),
    _templateObject9 = _taggedTemplateLiteral(['\n  font-size: 14px;\n  font-weight: 400;\n  line-height: 24px;\n'], ['\n  font-size: 14px;\n  font-weight: 400;\n  line-height: 24px;\n']),
    _templateObject10 = _taggedTemplateLiteral(['\n  font-size: 14px;\n  font-weight: 500;\n  line-height: 24px;\n'], ['\n  font-size: 14px;\n  font-weight: 500;\n  line-height: 24px;\n']),
    _templateObject11 = _taggedTemplateLiteral(['\n  font-size: 12px;\n  font-weight: 400;\n  line-height: 1;\n'], ['\n  font-size: 12px;\n  font-weight: 400;\n  line-height: 1;\n']);

function _taggedTemplateLiteral(strings, raw) {
    return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } }));
}

var Display4 = exports.Display4 = _styledComponents2.default.h1(_templateObject);
var Display3 = exports.Display3 = _styledComponents2.default.h1(_templateObject2);
var Display2 = exports.Display2 = _styledComponents2.default.h1(_templateObject3);
var Display1 = exports.Display1 = _styledComponents2.default.h1(_templateObject4);
var Headline = exports.Headline = _styledComponents2.default.h2(_templateObject5);
var Title = exports.Title = _styledComponents2.default.h3(_templateObject6);
var Subhead = exports.Subhead = _styledComponents2.default.h4(_templateObject7);
var Subhead2 = exports.Subhead2 = _styledComponents2.default.h4(_templateObject8);
var Body = exports.Body = _styledComponents2.default.p(_templateObject9);
var Body2 = exports.Body2 = _styledComponents2.default.p(_templateObject10);
var Caption = exports.Caption = _styledComponents2.default.span(_templateObject11);
